Sources/ApolloCodegenLib/Templates/TemplateRenderer.swift

```diff
@@ -160,7 +162,7 @@ private struct ImportStatementTemplate {
     private static func shouldImportSchemaModule(
       _ config: ReferenceWrapped<ApolloCodegenConfiguration>
     ) -> Bool {
-      config.output.operations != .inSchemaModule && config.output.schemaTypes.isInModule
+      config.output.operations != .inSchemaModule
```

Comment:
Schema types always end up in some kind of module, whether it's one we create or on that we link to, but as soon as operation files are not part of that namespace the schema module must be imported.